backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/optimizations/EscapeAnalysis.kt

```diff
@@ -1676,12 +1674,17 @@ internal object EscapeAnalysis {
         } catch (t: Throwable) {
             val extraUserInfo =
                         """
-                        Please try to disable escape analysis and rerun the build. To do it add "-Xdisable-phases=EscapeAnalysis" to the compilation command line or for Gradle, as an example:
+                        Please try to disable escape analysis and rerun the build. To do it add the following snippet to the gradle script:
 
-                            macosX64("macos") {
-                                compilations["main"].kotlinOptions.freeCompilerArgs += "-Xdisable-phases=EscapeAnalysis"
+                            kotlin {
```

Comment:
The message looks good to me but here is a couple of minor (= not necessary) remarks:

1. We can replace the `kotlin`-block with direct access to `kotlin.targets` to make the snippet more compact:

```gradle.kts
kotlin.targets.withType<org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget> {
    binaries.all {
        freeCompilerArgs += "-Xdisable-phases=EscapeAnalysis"
    }
}
```

2. The Groovy snippet will look like this:

```gradle
kotlin.targets.withType(org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget) {
    binaries.all {
        freeCompilerArgs += "-Xdisable-phases=EscapeAnalysis"
    }
}
   
```
Looks very close to the Kotlin snippet, so I'm not sure if we need to provide both.

3. This snippet will disable the escape analysis for one module only. If we want to disable it in the whole build, we need to add this to the root buildscript:

```gradle.kts
allprojects {
    val kotlinExtension = extensions.findByName("kotlin") as? org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
    kotlinExtension?.targets?.withType<org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget> {
        binaries.all {
            freeCompilerArgs += "-Xdisable-phases=EscapeAnalysis"
        }
    }
}
```
But I think that disabling it for only one module is enough.